packages/@aws-cdk/aws-lambda/lib/function.ts

```diff
@@ -856,6 +873,10 @@ Environment variables can be marked for removal when used in Lambda@Edge by sett
       throw new Error('Cannot configure \'securityGroup\' or \'allowAllOutbound\' without configuring a VPC');
     }
 
+    if (!props.vpc && props.filesystem) {
+      throw new Error('Cannot configurea \'filesystem\' without configuring a VPC.');
+    }
```

Comment:
Move this validation into the `if (props.filesystem)` conditional as well.

Reply:
That would be an error, I'm afraid. It would put the check after:
https://github.com/aws/aws-cdk/blob/46a16312825cc7b5bf985bc8b69146a1397690d0/packages/%40aws-cdk/aws-lambda/lib/function.ts#L859

which would cause the condition to never be true.

Reply:
> I'm finding a bit difficult to understand what you're saying here. It's not clear to me why the configureVpc() method has anything to do with this.

The code in question is within `configureVpc()`.... but I digress. Sure, I'll move the check & error to the conditional where the dependencies are being added. 

> Perhaps I didn't explain the intention behind my comment - I wanted to keep the validation around filesystem, in the same place(s) as where the filesystem is being configured. configureVpc() feels like the wrong place.

Yes. Starting with that would have been useful. :-)